### PR TITLE
Add warning for PreferNativeArm64

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
PreferNativeArm64 is not supported by .Net Apps. 
This code generates warning if PreferNativeArm64 set explicitly as true.

Related to: https://github.com/dotnet/msbuild/pull/10134

### CODE REVIEW

File: src/Tasks/Common/Resources/Strings.resx [hunk 1]
```diff
@@ -856,6 +856,10 @@ You may need to build the project on another operating system or architecture, o
     <value>NETSDK1189: Prefer32Bit is not supported and has no effect for netcoreapp target.</value>
     <comment>{StrBegin="NETSDK1189: "}</comment>
   </data>
+  <data name="PreferNativeArm64IgnoredForNetCoreApp" xml:space="preserve">
+    <value>NETSDK1214: PreferNativeArm64 is not supported and has no effect for netcoreapp target.</value>
+    <comment>{StrBegin="NETSDK1214: "}</comment>
+  </data>
```
Comment (ladipro) [inline src/Tasks/Common/Resources/Strings.resx:862]: nit: Move to the end of the file so the strings stay sorted by error code?

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets [hunk 2]
```diff
@@ -284,6 +284,19 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <Target Name="_CheckAndUnsetUnsupportedPreferNativeArm64"
+          BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
+          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(_TargetFrameworkVersionWithoutV)' >= '7.0'">
```
Comment (ladipro) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets:289]: I don't think this should be behind a version check. For `Prefer32Bit` it was likely done for backward compatibility but `PreferNativeArm64` is a brand new flag so we can disallow for any version of .NET.

**Review by ladipro — APPROVED** [review]

**Review by maridematte — APPROVED** [review]

**Review by rainersigwald — APPROVED** [review]
Wording for the error should probably get a thumbs-up from @baronfel.

File: src/Tasks/Common/Resources/Strings.resx [hunk 3]
```diff
@@ -948,5 +948,9 @@ You may need to build the project on another operating system or architecture, o
     <value>NETSDK1213: Targeting .NET 8.0 or higher in Visual Studio 2022 17.7 is not supported.</value>
     <comment>{StrBegin="NETSDK1213: "}</comment>
   </data>
+  <data name="PreferNativeArm64IgnoredForNetCoreApp" xml:space="preserve">
+    <value>NETSDK1214: PreferNativeArm64 is not supported and has no effect for netcoreapp target.</value>
```
[the remark this answers is not in the document]
Comment (baronfel) [reply src/Tasks/Common/Resources/Strings.resx:952]: The rephrasing that @rainersigwald is fine by me. We don't have consensus around how to discuss 'modern .NET' in contrast to '.NET Framework' and this is clear in comparison.

One thing I do slightly worry about is tech debt here - will we want to update the '6+' to '8+' when 6 goes out of support? If so I might just say ".NET Core" here to emphasize the difference (as opposed to bare ".NET").

**Pull request merged**